Assets/Scripts/Layers/LayerTypes/HierarchicalObjectLayerGameObject.cs

```diff
@@ -68,62 +75,72 @@ protected void OnDestroy()
+            if (transformProperty == default(TransformLayerPropertyData)) return;
+            
+            // unsubscribe events from previous property object, resubscribe to new object at the end of this if block
+            if (transformPropertyData != null) 
             {
-                if (transformPropertyData != null) //unsubscribe events from previous property object, resubscribe to new object at the end of this if block
-                {
-                    transformPropertyData.OnPositionChanged.RemoveListener(UpdatePosition);
-                    transformPropertyData.OnRotationChanged.RemoveListener(UpdateRotation);
-                    transformPropertyData.OnScaleChanged.RemoveListener(UpdateScale);
-                }
-
-                this.transformPropertyData = transformProperty; //take existing TransformProperty to overwrite the unlinked one of this class
-
-                UpdatePosition(this.transformPropertyData.Position);
-                UpdateRotation(this.transformPropertyData.EulerRotation);
-                UpdateScale(this.transformPropertyData.LocalScale);
-
-                transformPropertyData.OnPositionChanged.AddListener(UpdatePosition);
-                transformPropertyData.OnRotationChanged.AddListener(UpdateRotation);
-                transformPropertyData.OnScaleChanged.AddListener(UpdateScale);
+                transformPropertyData.OnPositionChanged.RemoveListener(UpdatePosition);
+                transformPropertyData.OnRotationChanged.RemoveListener(UpdateRotation);
+                transformPropertyData.OnScaleChanged.RemoveListener(UpdateScale);
             }
+
+            // take existing TransformProperty to overwrite the unlinked one of this class
+            this.transformPropertyData = transformProperty;
+
+            UpdatePosition(this.transformPropertyData.Position);
+            UpdateRotation(this.transformPropertyData.EulerRotation);
+            UpdateScale(this.transformPropertyData.LocalScale);
+            
+            // Reset the previous[X] to prevent a reset of the  
+            previousPosition = transformPropertyData.Position;
+            previousRotation = Quaternion.Euler(transformPropertyData.EulerRotation);
+            previousScale = transformPropertyData.LocalScale;
+
+            transformPropertyData.OnPositionChanged.AddListener(UpdatePosition);
+            transformPropertyData.OnRotationChanged.AddListener(UpdateRotation);
+            transformPropertyData.OnScaleChanged.AddListener(UpdateScale);
         }
 
         private void Update()
         {
             // We cannot use transform.hasChanged, because this flag is not correctly set when adjusting this transform using runtimeTransformHandles, instead we have to compare the values directly
             // Check for position change
-            if (transform.position != previousPosition)
+            if (worldTransform.Coordinate.ToUnity() != previousPosition.ToUnity())
```

Comment:
kan er niet een directe coordinate vergelijking gedaan worden (of worden toegevoegd)? dit lijkt me een beetje vatbaar voor floating point errors

Reply:
De Coordinate class heeft geen Equality operator implementatie; daar liep ik tegenaan toen ik dit stuk omschreef. Maar ik wilde de scope ook niet heel erg vergroten door operator overrides in de Coordinate converter package toe te voegen